programmieren-lernen/dependency-injection.md

```diff
@@ -6,26 +6,32 @@ alias: ["di"]
 Dependency Injection by Example:
 ```java
 public class MeineApp {
+  // hier übergibt man die Instanz der MeineApp Klasse
   private final DepA a = new DepA(this);
-  private final Depb b = new DepB(a);
+  // die DepB Klasse greift auf die DepA Klasse zu
+  private final DepB b = new DepB(a);
+  // Info: man muss nicht die Variablen hier ↑ initialisieren, geht auch in der onEnable.
   public void start(String[] args) {
-    // Do Stuff
+    b.doThat();
   }
-  // Getters here
 }
-```
-```java
+``````java
 public class DepA {
   private final MeineApp app;
+  // hier kann man dann die Instanz der MeineApp Klasse verwenden.
   public DepA(MeineApp app) { this.app = app; }
   public void doThis() {}
 }
-```
-```java
+``````java
 public class DepB {
   private final DepA a;
+  // da die DepB Klasse nur die Funktionalität der DepA Klasse braucht,
+  // übergibt man nur sie im Konstruktor
```

Comment:
```suggestion
  // Die DepB Klasse benötigt Zugriff auf die DepA Instanz.
  // Daher übrgibt man diese im Konstruktor
```

Weird wording. It does not require the functionality of the class, but access to the instance for whatever reason